sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala

```diff
@@ -642,7 +642,7 @@ class LogicalPlanToSQLSuite extends SQLBuilderTest with SQLTestUtils {
     checkColumnNames(
       """SELECT x.a, y.a, x.b, y.b
         |FROM (SELECT 1 AS a, 2 AS b) x
-        |INNER JOIN (SELECT 1 AS a, 2 AS b) y
+        |CROSS JOIN (SELECT 1 AS a, 2 AS b) y
```

Comment:
The join condition is eliminated because it is "trivial" according to the foldable propagation rule. This is annoying, but an expected by-product of detecting joins in the optimization phase. I can check to see if there is a way to split up the operator optimization batch so the cartesian check can be inserted before the constant propagation.